src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -192,7 +209,31 @@ public override void Emit(OpCode opcode, string str)
         public override void Emit(OpCode opcode, ConstructorInfo con) => throw new NotImplementedException();
         public override void Emit(OpCode opcode, Label label) => throw new NotImplementedException();
         public override void Emit(OpCode opcode, Label[] labels) => throw new NotImplementedException();
-        public override void Emit(OpCode opcode, LocalBuilder local) => throw new NotImplementedException();
+        public override void Emit(OpCode opcode, LocalBuilder local)
+        {
+            ArgumentNullException.ThrowIfNull(local);
+
+            if (local.Method != _methodBuilder)
+            {
+                throw new ArgumentException(SR.Argument_UnmatchedMethodForLocal, nameof(local));
+            }
+
+            int tempVal = local.LocalIndex;
+            if (opcode.Name!.StartsWith("ldloca"))
```

Comment:
This statement has a `!`, but the others don't. Is this intended? If so can we add a comment as to the why here? I assume to keep the compiler quiet, but it is a weird thing to have in one but no the others.

Reply:
Yes, it is for handling nullability warning, the null forgiving operator `!` not only shuts down the compiler, but it also tells the nullability analyzer no need to warn for `opcode.Name` further, this is by design, no need a comment.

Reply:
Let me propose a different approach to make this clearer. Create a local that performs the `!` and then use the local throughout.

```csharp
string name = opcode.Name!;
if (name.StartsWith("ldloca"))
...
```